tensorly/regression/cp_plsr.py

```diff
@@ -0,0 +1,230 @@
+from ..tenalg import khatri_rao, mode_dot, multi_mode_dot
+from ..cp_tensor import CPTensor, cp_to_tensor, cp_to_vec
+from ..decomposition import tucker
+from .. import backend as T
+from .. import unfold
+
+# Author: Cyrillus Tan, Jackson Chin, Aaron Meyer
+
+# License: BSD 3 clause
+
+
+class CP_PLSR:
+    """CP tensor regression
+
+        Learns a low rank CP tensor weight
+
+    Parameters
+    ----------
+    n_components : int
+        rank of the CP decomposition of the regression weights
+    tol : float
+        convergence value
+    n_iter_max : int, optional, default is 100
+        maximum number of iteration
+    random_state : None, int or RandomState, optional, default is None
+    verbose : bool, default is False
+        whether to be verbose during fitting
+    """
+
+    def __init__(
+        self, n_components, tol=1.0e-7, n_iter_max=100, random_state=None, verbose=False
+    ):
+        self.n_components = n_components
+        self.tol = tol
+        self.n_iter_max = n_iter_max
+        self.random_state = random_state
+        self.verbose = verbose
+
+    def get_params(self, **kwargs):
+        """Returns a dictionary of parameters"""
+        params = ["n_components", "tol", "n_iter_max", "random_state", "verbose"]
+        return {param_name: getattr(self, param_name) for param_name in params}
+
+    def set_params(self, **parameters):
+        """Sets the value of the provided parameters"""
+        for parameter, value in parameters.items():
+            setattr(self, parameter, value)
+        return self
+
+    def fit(self, X, Y):
+        """Fits the model to the data (X, Y)
+
+        Parameters
+        ----------
+        X : ndarray
+            tensor data of shape (n_samples, N1, ..., NS)
+        Y : 2D-array of shape (n_samples, n_predictions)
+            labels associated with each sample
+
+        Returns
+        -------
+        self
+        """
+        ## PREPROCESSING
+        # Check that both tensors are coupled along the first mode
+        if T.shape(X)[0] != T.shape(Y)[0]:
+            raise ValueError(
+                "The first modes of X and Y must be coupled and have the same length."
```

Comment:
Instead of "same length" would it be correct to say "same mode 0 length"?

Reply:
They should be synonyms. I said "the first modes" which means mode 0 in this case. You can change it to the other way if it makes more sense.